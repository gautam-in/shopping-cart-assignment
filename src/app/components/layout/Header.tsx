import React from "react";
import Link from "next/link";
import Image from "next/image";
import { useSelector, useDispatch } from "react-redux";

const Header = () => {
  let cartState = useSelector((state: any) => state.cart);
  console.log("cartState  from store : ", cartState);
  let cartText = "-";
  if (cartState) {
    cartText = `${cartState.totalItems} ${
      cartState.totalItems > 1 ? "items" : "item"
    }`;
  }

  return (
    <nav className=" h-28 shadow-lg">
      <div className="max-w-screen-lg mx-auto h-full flex justify-between items-center">
        <Link href="/offers" className="">
          <Image
            src="/static/images/logo_2x.png"
            width="150"
            height="150"
            alt="Logo Image"
          />
        </Link>
        <div className="hidden h-full md:flex lg:flex">
          <ul className="flex flex-col md:flex-row lg:flex-row h-full gap-4 items-end pb-4 text-gray-500 cursor-pointer  font-extrabold">
            <li>
              <Link href="/offers"> Home </Link>
            </li>
            <li>
              <Link href="/products">Products</Link>
            </li>
          </ul>
        </div>

        <div className="h-full flex flex-col pt-5 ">
          <ul className=" flex gap-4 mb-2 cursor-pointer">
            <li>
              <Link href="/login">SignIn</Link>
            </li>
            <li>
              <Link href="/logout">SignOut</Link>
            </li>
            <li>
              <Link href="/signup">Register</Link>
            </li>
            <li>
              <button
                data-collapse-toggle="mobile-menu"
                type="button"
                className="lg:hidden md:hidden inline-flex items-center p-2 ml-1 text-sm text-gray-500 rounded-lg hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200"
              >
                <span className="sr-only">Open main menu</span>
                <svg
                  className="w-6 h-6"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    fill-rule="evenodd"
                    d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                    clip-rule="evenodd"
                  ></path>
                </svg>
              </button>
            </li>
          </ul>
          <div className=" bg-[#ddd] flex-1 text-center py-2 cursor-pointer">
            <Image
              src="/static/images/cart1.svg"
              alt="cart"
              height={40}
              width={40}
              className="inline-block"
            />
            <span className="inline-block pl-2">{cartText}</span>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Header;
