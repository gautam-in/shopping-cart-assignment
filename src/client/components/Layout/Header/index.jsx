import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';
import withStyles from 'isomorphic-style-loader/withStyles';

import images from '../../../constants/images';
import WEB_PATH from '../../../constants/webPath';
import MiniCart from '../../MiniCart';
import styles from './Header.scss';

const Header = () => {
  const location = useLocation();
  const miniCart = useSelector((state) => state.miniCart);
  const [cartOpen, setCartOpen] = useState(false);

  const openCart = (e) => {
    e.preventDefault();
    setCartOpen(true);
  };

  return (
    <header>
      <nav>
        <div className="main-nav">
          <img src={images.logoImage} alt="Sabka Bazar" className="logo" />
          <ul className="page-nav">
            <li className={location.pathname === WEB_PATH.HOME ? ' active' : ''}>
              <Link to={WEB_PATH.HOME}>Home</Link>
            </li>
            <li className={location.pathname === WEB_PATH.PRODUCTS ? ' active' : ''}>
              <Link to={WEB_PATH.PRODUCTS}>Products</Link>
            </li>
          </ul>
        </div>
        <div className="action-nav">
          <ul className="action-nav-list">
            <li>
              <Link to={WEB_PATH.LOGIN}>Signin</Link>
            </li>
            <li>
              <Link to={WEB_PATH.SIGNUP}>Register</Link>
            </li>
          </ul>
          <div className="cart-box">
            <Link to={WEB_PATH.HOME} className="icon-cart cart-link" aria-label="cart" onClick={openCart} aria-labelledby="cart-modal-container" />
            <span>{`${miniCart.totalItems} items`}</span>
          </div>
          <MiniCart
            isOpen={cartOpen}
            toggle={setCartOpen}
            miniCart={miniCart}
          />
        </div>
      </nav>
    </header>
  );
};

export default withStyles(styles)(Header);
