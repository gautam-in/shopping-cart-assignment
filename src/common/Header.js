import React from 'react'
import { NavLink } from "react-router-dom";
import { useNavigate } from "react-router-dom";

function Header() {
    let navigate = useNavigate(); 

    return (
        <header className='header pg-lr-sp'>
            <div className='section1 fx_aE'>
                <img src="static/images/logo.png" className='logo' alt="sabka-bazar-logo"  onClick={()=>{navigate('Home');}}/>
                <nav className="nav-section">
                    <NavLink
                        to="/Home"
                        className="nav-link">
                        Home
                    </NavLink>
                    &emsp;
                    &emsp;
                    <NavLink
                        to="/Products"
                        className="nav-link">
                        Products
                    </NavLink>
                </nav>
            </div>
            <section className='section2'>
                <div className='auth'>
                    <nav className="nav-section">
                        <NavLink
                            to="/"
                            className="nav-link">
                            SignIn
                        </NavLink>
                        &emsp;
                        <NavLink
                            to="/Register"
                            className="nav-link">
                            Register
                        </NavLink>
                    </nav>
                    {/* <span>SignIn</span>
                    &emsp;
                    <span>Register</span> */}
                </div>
                <div className='header-cart'>
                    <img src="static/images/cart.svg" className='cartIcon' alt="cart-logo" />
                    &nbsp;
                    <span>0</span>
                    &nbsp;
                    <span>items</span>
                </div>
            </section>
        </header >
    )
}

export default Header