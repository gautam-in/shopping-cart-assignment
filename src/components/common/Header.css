.nav-bar {
    display: grid;
    grid-template-columns: repeat(1, 40% 40% auto);
    height: 100%;
    flex-direction: column;
}

.menu {
    display: flex;
    margin-top: 8%;
    margin-left: 11%
}

#products {
    margin-left: 30%;
}

#logo {
    margin-top: 1.5%;
    max-width: 180px;
    width: auto;
    margin-left: 35%;
}

.other-links {
    margin-top: 3%;
    display: flex;
}

.header {
    position: static;
    box-shadow: 0 8px 6px -6px black;
    width: 100% !important
}

.MuiGrid-spacing-xs-3>.MuiGrid-item {
    padding: 0px !important;
}

#items-count {
    color: 'black';
    font-size: 0.9rem;
    font-family: Dosis;
}

#icon-btn {
    background-color: #e7e7ec;
    max-width: 100px;
    max-height: 100%;
    border-radius: 0%;
    border: none;
    margin-top: 7%;
}

.cart {
    width: 30%;
    height: 85%;
    background-color: white;
    outline: 0;
    margin: auto;
    margin-top: 6%;
    margin-left: 52.8%;
}

.cart-items {
    height: '72%';
    overflow: 'hidden';
    overflow-y: 'auto';
}

#minus-btn {
    border-radius: 50%;
    border: none;
    background-color: #f50057;
    color: white;
    display: block;
    height: 100%;
    width: 120%;
    outline: none;
}

#plus-btn {
    border-radius: 50%;
    border: none;
    background-color: #f50057;
    color: white;
    display: block;
    height: 100%;
    width: 80%;
    outline: none
}

@media only screen and (max-width:600px) {
    .menu {
        display: none;
    }
    .other-links {
        display: none;
    }
    .nav-bar {
        display: grid;
        grid-template-columns: repeat(1, 50% 0% 50%);
    }
    #logo {
        margin-left: 0%;
        max-width: 100px;
        width: auto;
    }
    #icon-btn {
        background-color: #e7e7ec;
        margin-left: 50%;
        margin-top: 0%;
    }
    .cart {
        width: 98%;
        height: 100%;
        margin-top: 0%;
        margin-left: 0%;
    }

    #plus-btn {
        border-radius: 50%;
        border: none;
        background-color: #f50057;
        color: white;
        display: block;
        height: 100%;
        width: 90%;
        outline: none
    }
}