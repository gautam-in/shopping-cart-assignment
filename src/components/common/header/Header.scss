$grey: #eaeaea;
$darkGrey: #acacac;
$lightBlue: #50dae4;
$pink: #d10157;

.headerContainer {
  box-shadow: 0 5px 10px -1px $grey;
  display: flex;
  justify-content: space-between;
  padding: 10px 5%;
  img {
    width: 150px;
  }
  .link {
    &.active {
      color: #000;
    }
    text-decoration: none;
    color: $darkGrey;
    padding: 10px 10px;
    margin: 0px 10px;
  }
  .leftContainer {
    position: relative;
    .link {
      color: #000;
      font-size: 12px;
      padding: 0px 5px;
      margin: 0px 5px;
    }
    .cartContainer {
      position: absolute;
      background-color: #eaeaea;
      top: 30px;
      padding: 5px;
      display: flex;
      align-items: center;
      img {
        width: 40px;
      }
    }
  }
  .rightContainer {
    display: flex;
    align-items: flex-end;
  }
  button {
    &:focus {
      outline: 0;
    }
    background: transparent;
    border: 0;
    opacity: 0.5;
    color: #fff;
    height: 40px;
    cursor: pointer;
  }
}

@media screen and (max-width: 991px) {
  .headerContainer {
    padding: 10px;
  }
}
