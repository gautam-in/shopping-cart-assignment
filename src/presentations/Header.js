import React from 'react';
import {Link} from 'react-router';
import {connect} from 'react-redux';
// import {push} from 'react-router-redux';
// import Logo from '../../static/images/logo.png';
import {getViewPortDimensions} from '../utils';
import {resetCartData} from '../actions/productActions';
// import CartIcon from '../../static/images/cart.svg';//'';
import LazyLoad from 'react-lazyload';
class Header extends React.Component {
     constructor(props) {
        super(props);
        this.state = {
            logo: '',
            clickable: false,
            userName:'',
            itemCount: 0,
            dimensions:{}
        }
        // this.clickEvent = this.clickEvent.bind(this);
        // this.showHome = this.showHome.bind(this);
        // this.doResizeActions = this.doResizeActions.bind(this);
        // this.cartClick = this.cartClick.bind(this);
        // this.resetCart = this.resetCart.bind(this);
        // this.resize = this.resize.bind(this);
    }
    componentDidMount(){
        if(this.props.signin.userData && this.props.signin.userData.userEmail){
            this.setState({userName:this.props.signin.userData.userEmail});
        } else if(this.props.signup && this.props.signup.userData && this.props.signup.userData.userEmail){
            this.setState({userName:this.props.signup.userData.userEmail});
        }
        this.doResizeActions();
        window.addEventListener('resize',this.resize,false);
    }
    componentWillUnmount(){
        window.removeEventListener('resize',this.resize,false);
    }

    static getDerivedStateFromProps(props,state){
        if(props.cartInfo){
            state.itemCount= props.cartInfo.length;
        }
        if(props.productInfo && props.productInfo.cartItems){
            state.itemCount= props.productInfo.cartItems.length
        }
        return state;
    }
    resize=()=>{
        this.doResizeActions();
    }
    resetCart=()=>{
        this.props.resetCartData();
    }
    doResizeActions=()=>{
        let dimensions = getViewPortDimensions();
        this.setState({dimensions: dimensions});
    }
    cartClick=()=>{
        if(this.state.dimensions && this.state.dimensions.width && this.state.dimensions.width < 480){ //1024
            this.props.router.push('/cart');
        } else {
            this.props.cartClick();
        }
    }
    clickEvent=(event)=>{}
    showHome=()=>{
        this.props.router.push('/');
    }
    render(){   
        const {userName,itemCount} = this.state;
        return (<div className="header-area">
            <header>
                <nav className="sticky-nav">
                    <div className="row">
                        <img onClick={this.showHome} src={window.location.origin + '/static/images/logo.png'} alt="logo" className="logo" /> {/* {Logo} */}
                        <ul className="main-nav js--main-nav">
                            <li><Link to={`/home`} >Home</Link></li>
                            <li><Link to="/products">Products</Link></li>
                            <div className="logo-icon-header">
                            <li><Link to={`/home`} ><i className="ion-ios-home"></i></Link></li>
                            <li><Link to="/products"><i className="ion-ios-list"></i></Link></li>
                            <li><Link to={`/signin`} ><i className="ion-ios-person"></i></Link></li>
                            <li><Link to="/signup"><i className="ion-ios-personadd"></i></Link></li>
                            </div>
                           
                        </ul>
                        <div className="guest-info">
                            <div>
                            {userName && userName.length ? (<div> 
                                <p className="useremail" title={userName} aria-label={userName}>{userName}</p>
                            </div>) : (<ul className="main-nav js--main-nav">
                                <li onClick={this.resetCart}><Link to="/signin">Signin</Link></li>
                                <li onClick={this.resetCart}><Link to="/signup">Register</Link></li>
                            </ul>)}
                            </div>
                            <div className="cartArea" aria-labelledby="cart" onClick={this.cartClick}>
                                <div className="cart-icon">
                                    <LazyLoad height={25} once>
                                        <img src={window.location.origin + '/static/images/cart.svg'} className="fill-purple"></img>
                                    </LazyLoad>
                                </div>
                                <span>{itemCount} items</span>
                            </div>
                        </div>
                    </div>
                </nav>
            
        </header>
        </div>);
}}
function mapStateToProps(state){
    return{
        signin:state.signinReducer,
        signup:state.signUpReducer,
        productInfo: state.productReducer
    }
}
const mapDispatchToProps = (dispatch,ownProps) => {
    return {
        resetCartData: ()=>{dispatch(resetCartData())}
    }
}
export default connect(mapStateToProps,mapDispatchToProps)(Header);