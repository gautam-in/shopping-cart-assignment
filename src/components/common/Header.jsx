import React from 'react';
import AppBar from '@material-ui/core/AppBar'
import { Toolbar, Typography, Link, Modal, Grid, Button, Paper } from '@material-ui/core';
import ShoppingCartIcon from '@material-ui/icons/ShoppingCart';
import IconButton from '@material-ui/core/IconButton';
import { withRouter } from 'react-router-dom';
import CloseIcon from '@material-ui/icons/Close';
import { connect } from 'react-redux';
import { bindActionCreators } from "redux";
import addItem from '../../redux/actions/addItem';
import removeItem from '../../redux/actions/removeItem';
import './Header.css'
class Header extends React.Component {
    state = {
        open: false
    }
    processSabkaBazaarLogo = () => {
        return (
            <div className="img">
                <img id="logo"
                    src="/static/images/logo_2x.png"
                    alt="Sabka Bazaar" />
            </div>
        )
    }

    processMenuOptions = () => {
        return (
            <div className="menu">
                <div>
                    <p id="home">
                        <Link onClick={() => this.props.history.push("/")} style={{ color: "#52525d" }}>
                            <strong>Home</strong>
                        </Link>
                    </p>
                </div>

                <div>
                    <p id="products">
                        <Link onClick={() => this.props.history.push("/products")} style={{ color: "#52525d" }}>
                            <strong>Products</strong>
                        </Link>
                    </p>
                </div>

            </div>
        );
    }

    processOtherOptions = () => {
        return (
            <div>
                <div className="other-links">
                    <div>
                        <Link onClick={() => this.props.history.push("/signin")}
                            style={{ color: '#52525d' }}>
                            <strong>SignIn</strong>
                        </Link>
                    </div>
                    <div>
                        <Link onClick={() => this.props.history.push("/register")}
                            style={{ marginLeft: '15%', color: '#52525d' }}>
                            <strong>Register</strong>
                        </Link>
                    </div>
                </div>
                <div className="icon-class">
                    <IconButton onClick={this.setModalState} id="icon-btn">
                        <ShoppingCartIcon style={{ fontSize: "1.5rem" }} color="secondary" />
                        <span id="items-count">{`${this.props.productdetail.count} items`}</span>
                    </IconButton>
                </div>
            </div>
        );
    }

    handleClose = () => {
        this.setState({ open: false })
    }

    setModalState = () => {
        this.setState({ open: true })
    }


    processModal = () => {
        return (
            <Modal
                open={this.state.open}
                onClose={this.handleClose}
                aria-labelledby="simple-modal-title"
                aria-describedby="simple-modal-description"
            >
                {
                    <div className="cart">
                        {
                            this.props.productdetail.count ?
                                this.processCartView()
                                :
                                this.processEmptyCart()
                        }
                    </div>
                }
            </Modal>

        );
    }

    renderCartViewHeader = () => {
        console.log(this.props.productdetail)
        return <div style={{ height: '7%' }}>
            <Grid container style={{ backgroundColor: 'black', color: 'white', height: '100%' }}>
                <Grid item xs={10}>
                    <Typography style={{ padding: '1%' }} variant="subtitle1"><strong>{`My Cart (${this.props.productdetail.count} items)`}</strong></Typography>
                </Grid>
                <Grid item xs={2}>
                    <IconButton onClick={this.handleClose} style={{ marginTop: '-11%', color: 'white', float: 'right' }}>
                        <CloseIcon fontSize="inherit" style={{ color: 'white' }} />
                    </IconButton>
                </Grid>
            </Grid>
        </div>
    }
    renderItemDetails = () => {
        return this.props.productdetail.details.map(data => {
            return (
                <Paper key={data.id} style={{ margin: '2% 0%', boxShadow: 'none' }}>
                    <Grid container spacing={1}>
                        <Grid item xs={3} style={{ padding: '2%' }}>
                            <img style={{ width: '80%' }} src={data.imageURL} alt={data.name} />
                        </Grid>
                        <Grid item xs={9}>
                            <Typography variant="subtitle2"><strong>{data.name}</strong></Typography>
                            <Grid container style={{ margin: '1%' }}>
                                <Grid item xs={"auto"}>
                                    <button id="minus-btn"
                                        onClick={() => this.props.removeItem(data)}>
                                        -
                                             </button>
                                </Grid>
                                <Grid item xs={1} style={{ textAlign: 'center' }}>
                                    {data.count}
                                </Grid>
                                <Grid item xs={1}>
                                    <button id="plus-btn"
                                        onClick={() => this.props.addItem(data)}>
                                        +
                                            </button>
                                </Grid>
                                <Grid item xs={1}>
                                    X
                                </Grid>
                                <Grid item xs={6}>
                                    {`Rs.${data.price}`}
                                </Grid>
                                <Grid item xs={2}>
                                    {`Rs.${data.totalPrice}`}
                                </Grid>
                            </Grid>
                        </Grid>
                    </Grid>
                </Paper>
            );
        })
    }

    renderPromotionalTag = () => {
        return <Paper style={{ margin: '2%', padding: '1%', boxSizing: 'border-box' }}>
            <Grid container style={{ height: '100%' }} spacing={1}>
                <Grid item xs={3}>
                    <img src="/static/images/lowest-price.png" alt="lowest price" style={{ width: '100%' }} />
                </Grid>
                <Grid item xs={9}>
                    <span style={{ marginTop: '2vh' }}>You won't find it cheaper anywhere</span>
                </Grid>
            </Grid>
        </Paper>
    }

    renderProceedToCheckout = () => {
        return <>.
            <Paper style={{ padding: '4%' }}>
                <div style={{ marginBottom: '1.5%', textAlign: 'center' }}>Promo code can be applied on payment page</div>
                <Button fullWidth
                    onClick={this.handleClose}
                    style={{ textTransform: 'none', fontWeight: 'bolder' }} variant="contained" color="secondary">
                    Proceed to checkout
                        <span style={{ color: 'white', marginLeft: 'auto', fontWeight: 'bolder' }}>
                        {`Rs. ${this.props.productdetail.totalAmt}`}
                    </span>
                    <span style={{ color: 'white', marginLeft: '2%', marginTop: "-0.5%", fontWeight: '1000' }}>
                        {`>`}
                    </span>
                </Button>
            </Paper>
        </>

    }
    processCartView = () => {
        return (
            <div style={{ backgroundColor: '#e7e7ec', height: '100%' }}>
                {this.renderCartViewHeader()}
                <div className="cart-items">
                    {this.renderItemDetails()}
                    {this.renderPromotionalTag()}
                </div>
                {this.renderProceedToCheckout()}
            </div>

        )
    }

    processEmptyCart = () => {
        return (
            <>
                <div style={{ height: '39%' }}>
                    <Grid container style={{ backgroundColor: 'black', color: 'white' }}>
                        <Grid item xs={10}>
                            <Typography style={{ padding: '2%' }} variant="h6"><strong>My Cart</strong></Typography>
                        </Grid>
                        <Grid item xs={2}>
                            <IconButton onClick={this.handleClose} style={{ color: 'white', float: 'right' }}>
                                <CloseIcon fontSize="inherit" style={{ color: 'white' }} />
                            </IconButton>
                        </Grid>
                    </Grid>
                </div>
                <div style={{ textAlign: 'center', height: '50%' }}>
                    <Typography variant="h6"><strong>No items in your cart</strong></Typography>
                    <div>
                        Your favourite items are just a click away
                    </div>
                </div>
                <div style={{ padding: '1%', height: '10%' }}>
                    <Button fullWidth
                        onClick={this.handleClose}
                        style={{ textTransform: 'none', borderRadius: '0%' }} variant="contained" color="secondary">Start Shopping</Button>
                </div>
            </>
        );
    }

    render() {
        return (
            <header className="header">
                <div className="nav-bar">
                    <div>
                        {this.processSabkaBazaarLogo()}
                    </div>
                    <div>
                        {this.processMenuOptions()}
                    </div>
                    <div>
                        {this.processOtherOptions()}
                    </div>
                    {this.processModal()}
                </div>
            </header >
        )
    }
}



const mapStateToProps = (state) => {
    return {
        productdetail: state.productdetail,
    }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
        addItem,
        removeItem
    },
        dispatch)
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));