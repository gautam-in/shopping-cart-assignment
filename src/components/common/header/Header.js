import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { setActiveTab } from "../../../redux/actions/actions";
import { useWindowSize } from "../utils";
import "./Header.scss";

const Header = (props) => {
  const dispatch = useDispatch();
  const size = useWindowSize();
  const login = useSelector((state) => state.login);
  const activeTab = useSelector((state) => state.activeTab);

  const clickHandler = (activeTab) => {
    dispatch(setActiveTab(activeTab));
  };

  return (
    <header className="headerContainer">
      <img src={"/images/logo.png"} alt="logo" />
      {size.width >= 750 && (
        <div className="rightContainer">
          <Link
            className={activeTab === "Home" || login ? "active link" : "link"}
            to="/home"
            onClick={() => clickHandler("Home")}
          >
            Home
          </Link>
          <Link
            className={activeTab === "Products" ? "active link" : "link"}
            to="/productsPage"
            onClick={() => clickHandler("Products")}
          >
            Products
          </Link>
        </div>
      )}
      <div className="leftContainer">
        <Link
          className="link"
          to="/"
          onClick={() => {
            dispatch(setActiveTab(""));
          }}
        >
          SignIn
        </Link>
        <Link
          className="link"
          to="/signUp"
          onClick={() => {
            dispatch(setActiveTab(""));
          }}
        >
          Register
        </Link>
        <div className="cartContainer">
          <button onClick={props.handleCartClick}>
            <img src={"/images/cart.svg"} alt="cart" />
          </button>
          <span>{props.totalCartItems} items</span>
        </div>
      </div>
    </header>
  );
};

export default Header;
