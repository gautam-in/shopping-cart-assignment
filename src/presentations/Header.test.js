import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Header from './Header';
import {getViewPortDimensions} from '../utils';
const buildStore = configureStore([thunk]);
Enzyme.configure({adapter: new Adapter(),disableLifecycleMethods: false,lifecycleExperimental: true,});

describe('Header render part',()=>{
    const cartClick = jest.fn();
    const props={
        productReducer:{cartItems:[]},
        cartClick,
        signinReducer:{userData:{userEmail:''}},
        signUpReducer:{userData:{userEmail:''}},
        router:{push:(data)=>{'/'+data+''}}
    };
    let wrapper,shallowWrapper,store;
    const initialState = {
        isMobileView: false,
        logo: '',
        clickable: false,
        userName:'',
        itemCount: 0,
        dimensions:getViewPortDimensions()
    };
    let spyCartClick;
    beforeAll(()=>{
        store = buildStore(props);
        shallowWrapper = shallow(<Provider store={store} {...props}>
          <Header />
        </Provider>);
        // spyCartClick = jest.spyOn(shallowWrapper.dive().dive().instance(),'cartClick');
        // props.router = shallowWrapper.instance().props.router;
        // store = buildStore(props);
    });
    beforeEach(() => {
    // fetch = global.fetch;
        wrapper = mount(<Provider store={store} {...props}>
                        <Header />
                        </Provider>);
        spyCartClick = jest.spyOn(wrapper.instance().props,'cartClick');
    });
    afterEach(() => {
        wrapper.unmount();
    });
    afterAll(()=>{
        shallowWrapper.unmount();
    });
    it('Renders Correctly',()=>{
        expect(shallowWrapper).toMatchSnapshot();
        // console.log(shallowWrapper.dive().dive().state());
    });
    it('initial state',()=>{
        expect(shallowWrapper.dive().dive().state()).toEqual(expect.objectContaining(initialState));
    });
    it('cart count',()=>{
        // console.log(shallowWrapper.dive().props().productInfo.cartItems);
        expect(shallowWrapper.dive().props().productInfo.cartItems).toBe(props.productReducer.cartItems);
    });
    // it('cart click',()=>{
    //     // wrapper.find('.cartArea').at(0).simulate('click');

    //     // expect(cartClick).toHaveBeenCalled();
    //     // expect(props.cartClick).toHaveBeenCalled(0);

    //     // expect(spyCartClick).toHaveBeenCalled(1);
    //     // console.log(wrapper.instance());
    //     // expect(shallowWrapper.dive().dive().instance().cartClick()).toHaveBeenCalled();
    // });
});