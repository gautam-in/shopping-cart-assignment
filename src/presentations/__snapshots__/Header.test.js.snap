// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render part Renders Correctly 1`] = `ShallowWrapper {}`;
